compiler/inferring/node-recalc.cpp

```diff
@@ -160,14 +159,11 @@ NodeRecalc::NodeRecalc(tinf::Node *node, tinf::TypeInferer *inferer) :
   inferer_(inferer) {}
 
 void NodeRecalc::on_changed() {
-  __sync_synchronize();
-
   //fmt_fprintf(stderr, "{} : {} -> {}\n", node_->get_description(), type_out(node_->get_type()), type_out(new_type()));
   node_->set_type(new_type_);
   new_type_ = nullptr;
 
-  __sync_synchronize();
```

Comment:
I'm not sure if this correct, even if it works on tests. As AutoLocker does nothing with std::atomic, happens-before can be not conducted here, and compiler could be allowed to rearrange code leading to something wrong. Probably, this wouldn't happen, but maybe it's better to wait untill normal lockers are used.

Reply:
Finally we decided, that __sync_synchronize() is not needed here